fuel-gson/src/main/kotlin/com/github/kittinunf/fuel/gson/FuelGson.kt

```diff
@@ -5,21 +5,26 @@ import com.github.kittinunf.result.Result
 import com.google.gson.Gson
 import com.google.gson.reflect.TypeToken
 import java.io.Reader
+import java.lang.reflect.Type
 
 /**
  * Created by ihor_kucherenko on 7/4/17.
  * https://github.com/KucherenkoIhor
+ *
+ * Note: Use the default constructor iff no generics
  */
-class GsonDeserializer<out T : Any> : ResponseDeserializable<T> {
+class GsonDeserializer<out T : Any>(private val type: Type) : ResponseDeserializable<T> {
+    override fun deserialize(reader: Reader): T = Gson().fromJson<T>(reader, type)
+}
 
-    override fun deserialize(reader: Reader): T
-            = Gson().fromJson<T>(reader, object : TypeToken<T>() {}.type)
+inline fun <reified T : Any> Request.responseObject(noinline handler: (Request, Response, Result<T, FuelError>) -> Unit) =
+        response(gsonDeserializerOf(), handler)
 
-}
+inline fun <reified T : Any> Request.responseObject(handler: Handler<T>) = response(gsonDeserializerOf(), handler)
+
+inline fun <reified T : Any> Request.responseObject() = response(gsonDeserializerOf())
 
-fun <T : Any> Request.responseObject(handler: (Request, Response, Result<T, FuelError>) -> Unit) = response(GsonDeserializer<T>(), handler)
+inline fun <reified T : Any> gsonDeserializerOf() = GsonDeserializer<T>(object : TypeToken<T>(){}.type)
```

Comment:
I wonder whether we actually need a full-fledged class for this? @yoavst Do you think we can just get away with using the `object : ResponseDeserializable` and get rid of the `GsonDeserializer` altogether?

Reply:
Changed

Reply:
Why it will caused by java.lang.IllegalStateException: Gson().fromJson<T>(reade…: TypeToken<T>() {}.type) must not be null ?
My class is data class ResponseString(val code: String, val time: String, val data: String),and the json is {"code":"0","time":"2018-11-27 15:01:04","data":"eyJ0eXAiO"}